docs/proposals/20250724-independent-worker-topology-versions.md

```diff
@@ -0,0 +1,500 @@
+  - i.e. you have AI training and inference in the same cluster. You may have a node group for inference and a node group for training. When you have high priority training it is allowed to consume a portion of the inference
+    nodes to improve the training speed and later that is returned to the inference nodes when training is complete
+- increased compute cost of additional control planes
+- shared services across a multi-tenant environment in same cluster
+- single cluster for configuration management
+  - advanced users know how to propogate configurations across clusters but it comes with additional complexity that some customers may be averse to
+
+## Upgrade Strategy
+
+If applicable, how will the component be upgraded? Make sure this is in the test plan.
+
+Consider the following in developing an upgrade strategy for this enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing cluster required to make on upgrade in order to keep previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing cluster required to make on upgrade in order to make use of the enhancement?
+
+## Additional Details
+
+### Test Plan
+
+#### Scenario: Existing behavior with ClusterClass Topology Version
+When a cluster admin updates the `spec.topology.version` of their existing cluster with *no* worker node group versions defined, the default behavior should be verified. 
+- Ensure rolling upgrades of control planes and Workers are updated to the desired version. 
+
+#### Scenario: Single cluster uses Worker Node Group Version for the First Time
+When a cluster admin adds a new version to a specific MachineDeployment/MachinePool Topology, the controller should: 
+- Validate that the upgrade is supported (allowable version skew)
+- Remediation should be paused **only** for the MachineDeployment/MachinePool Topology that was modified. 
+- A rolling upgrade of the target MachineDeployment/MachinePool should take place.
+- Remediation and other actions on control planes and MachineDeployment/MachinePools not managed by the version change should occur as usual. 
+
+#### Scenario: Cluster Version Topology Upgraded on a Cluster with Independent Worker Node Group Versions
+When a cluster admin updates the version on `spec.topology.version` on a cluster which has independent worker node group versions defined, the controller should: 
+- Validate that the upgrade is supported (allowable version skew).
+  - This now has to take into account that there may be none or many independent worker node group versions defined within the cluster and validate the skew on each one. 
+- Remediation should be paused **only** for worker node groups that use the cluster-level version (those that **do not** have independent worker node group versions defined). Control planes will by default use only the version specified in the `spec.topology.version`. 
```

Comment:
We validated this was the case that remediation is paused right? We took notes when testing this I think